src/librustdoc/passes/collect_trait_impls.rs

```diff
@@ -155,14 +153,13 @@ impl<'a, 'tcx> SyntheticImplCollector<'a, 'tcx> {
 impl<'a, 'tcx> DocFolder for SyntheticImplCollector<'a, 'tcx> {
     fn fold_item(&mut self, i: Item) -> Option<Item> {
         if i.is_struct() || i.is_enum() || i.is_union() {
-            if let (Some(hir_id), Some(name)) =
-                (self.cx.tcx.hir().as_local_hir_id(i.def_id), i.name.clone())
-            {
-                self.impls.extend(get_auto_traits_with_hir_id(self.cx, hir_id, name.clone()));
-                self.impls.extend(get_blanket_impls_with_hir_id(self.cx, hir_id, name));
-            } else {
-                self.impls.extend(get_auto_traits_with_def_id(self.cx, i.def_id));
-                self.impls.extend(get_blanket_impls_with_def_id(self.cx, i.def_id));
+            // FIXME(eddyb) is this `doc(hidden)` check needed?
```

Comment:
My guess would be "not needed" or "needed only as optimization", if an item is hidden its impls won't be shown either way.